### PR TITLE
Fix assignment of AInt to u32 in the expression generator

### PR DESCRIPTION
**Pull request by amaiorano**
Apply the same fix that was made for AFloat to AInt when emitting the value assignment WGSL. Otherwise, if
globalTestConfig.unrollConstEvalLoops is true, as it is on Windows, then we emit invalid WGSL because we index an array of AInts with a runtime index, forcing concretization, which results in an assignment of i32 to u32. The fix is to do as was done with AFloat: we emit direct assignments of the literal values into the output array. Also made sure we don't emit the array of values in this case, as it's unused.

This fixes `'webgpu:shader,execution,expression,unary,ai_assignment:*'` tests that were failing on Windows.

Issue: #1644 

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [x] Tests are properly located in the test tree.
- [x] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [x] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [x] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

File: src/webgpu/shader/execution/expression/expression.ts [hunk 1]
```diff
@@ -48,6 +48,13 @@ export type Config = {
   vectorize?: number;
 };
 
+// Returns true if types contains an abstract float or abtract int
+function isAbstract(types: Type[]) {
```
Comment (ben-clayton) [inline src/webgpu/shader/execution/expression/expression.ts:52]: Already exists here:
https://github.com/gpuweb/cts/blob/3f04968bcc3bacf13aabf3c631a9459ffaddf262/src/webgpu/util/conversion.ts#L1613

File: src/webgpu/shader/execution/expression/expression.ts [hunk 2]
```diff
@@ -601,7 +608,7 @@ function basicExpressionShaderBody(
     // Constant eval
     //////////////////////////////////////////////////////////////////////////
     let body = '';
-    if (parameterTypes.some(ty => scalarTypeOf(ty).kind === 'abstract-float')) {
+    if (isAbstract(parameterTypes)) {
```
Comment (ben-clayton) [inline src/webgpu/shader/execution/expression/expression.ts:611]:
```suggestion
    if (parameterTypes.some(isAbstractType)) {
```

File: src/webgpu/shader/execution/expression/expression.ts [hunk 3]
```diff
@@ -627,10 +634,16 @@ function basicExpressionShaderBody(
   }`;
     }
 
+    // If params are abstract, we will assign them directly to the storage array, so skip the values array.
+    let valuesArray = '';
+    if (!isAbstract(parameterTypes)) {
```
Comment (ben-clayton) [inline src/webgpu/shader/execution/expression/expression.ts:639]:
```suggestion
    if (parameterTypes.some(isAbstractType)) {
```

**Review by ben-clayton — APPROVED** [review]
Thanks!

**Comment by amaiorano** [comment]
> Thanks!

I'm still not used to GitHub. I had replied "Thanks!" to your suggetions, but those got lost when I resolved them...

**Pull request merged**